### PR TITLE
Add "Manage Premiums" to AdminUI, toward  dev/core#3912 .

### PR DESCRIPTION
**Pull request by twomice**
Overview
----------------------------------------
Add "Manage Premiums" to AdminUI

Before
----------------------------------------
before

After
----------------------------------------
afer

Technical Details
----------------------------------------
None.

Comments
----------------------------------------
None. Should be ready to review now.

### CODE REVIEW

**Comment by UshaMakoa** [comment]
Nice Thank you

But :

1. For the "add product" button in your SK/SearchDisplay (/civicrm/admin/search#/edit/26), in the toolbar, you can change the "Update Product" Action by "Other" and put this link : 'civicrm/admin/contribute/managePremiums?action=add&reset=1'
2. I'm not sure the Action menu is useful?
3. The links in the Menu (at the end of each row) doesn't function well : Preview, Edit, Delete (but Disable function) (you certainly could choose the link "Other"=
4. You can delete the help text (blue one) in the SK/SearchDisplay and leave the help text (green one) in the FB rich content but it would be better to ts() this text please. here is an example : 
```
<div>{{:: ts('The status rules provided by default may be sufficient for your organization. However, you can easily change the status names and\/or adjust the rules by clicking the Edit links below. Or you can add a new status and rule.') }}</div>
</div>
```

**Comment by twomice** [comment]
Thanks for the review @UshaMakoa . I've removed "WIP" ("work in progress") from the title, and I think now it's actually ready. With help from @aydun  I was able to address the items you mentioned, and a little more.

i suggest this is ready for review now.

**Comment by twomice** [comment]
Thanks for the help @aydun !

**Comment by colemanw** [comment]
@twomice thanks for this PR it looks very good.
Just one thing: our new convention is to declare afforms using php files instead of json so that strings can be translated. So `afsearchPremiumProducts.aff.json` should be renamed like https://github.com/civicrm/civicrm-core/pull/28499 - also the title should be wrapped in `E::ts()` and all the null values (and the modified_date) can be deleted for concision.

**Comment by twomice** [comment]
@colemanw I've made those changes now; hopefully this is good, but please let me know if more is needed. Thanks!

**Comment by aydun** [comment]
The paths in the XML and DAO are different: they should have the `/edit` part as in the DAO.  However, these are now updated in master except for the `preview` link.  Can you try rebasing on master and only adding the `preview` link?

**Comment by twomice** [comment]
@aydun I've made that change now (if I understand you correctly), and functionality seems to be correct. Thanks for the feedback! Any further requirements, please let me know.

File: xml/schema/Contribute/Product.xml [hunk 1]
```diff
@@ -6,6 +6,9 @@
   <comment>able - stores "product info" for premiums and can be used for non-incentive products </comment>
   <name>civicrm_product</name>
   <add>1.4</add>
+  <paths>
+    <preview>civicrm/admin/contribute/managePremiums?action=preview&amp;reset=1&amp;id=[id]</preview>
```
Comment (colemanw) [inline xml/schema/Contribute/Product.xml:10]: This path doesn't seem to match the one in the generated DAO file.

File: ext/civicrm_admin_ui/ang/afsearchPremiumProducts.aff.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+return [
+  'type' => 'search',
+  'title' => E::ts('Premium Products'),
+  'placement' => [],
+  'icon' => 'fa-list-alt',
+  'server_route' => 'civicrm/admin/contribute/managePremiums',
+  'is_public' => FALSE,
+  'permission' => [
+    'access CiviCRM',
```
Comment (colemanw) [inline ext/civicrm_admin_ui/ang/afsearchPremiumProducts.aff.php:13]: We need to match this to whatever permission was being checked previously for this screen.
I find the menu xml files a bit too magical in the way they assign permissions to paths, but I *think* the way it works is that the standard url inherits its permissions from the next path in the hierarchy that has them declared, and that would be `civicrm/admin`. And that one checks `administer CiviCRM system + administer CiviCRM data + access CiviCRM`. That seems a bit silly. Maybe we should set this to
```suggestion
    'access CiviContribute',
    'administer CiviCRM data',
```

**Comment by colemanw** [comment]
@twomice I think we're very close on this one. if you could just check those last 2 comments, I think that's all that's left to do before merging.

**Comment by eileenmcnaughton** [comment]
@colemanw if you want to merge this & make those changes in a follow up I'll merge the follow up

File: xml/schema/Contribute/Product.xml (hunk 1 above)
Comment (colemanw) [inline xml/schema/Contribute/Product.xml:10]:
```suggestion
    <preview>civicrm/admin/contribute/managePremiums/edit?action=preview&amp;reset=1&amp;id=[id]</preview>
```

**Comment by colemanw** [comment]
Ok I fixed the path. Lets merge this now & try to fix the Afform permissions during the RC period.

**Comment by twomice** [comment]
Thanks; sorry I couldn't get to this quicker.

**Pull request merged**